android/capacitor/src/main/java/com/getcapacitor/CapConfig.java

```diff
@@ -52,81 +116,341 @@ private void loadConfig(AssetManager assetManager) {
         }
     }
 
+    /**
+     * Deserializes the config from JSON into a Capacitor Configuration object.
+     */
+    private void deserializeConfig() {
+        // Server
+        html5mode = getBoolean(configJSON, "server.html5mode", html5mode);
+        serverUrl = getString(configJSON, "server.url", null);
+        hostname = getString(configJSON, "server.hostname", hostname);
+        androidScheme = getString(configJSON, "server.androidScheme", androidScheme);
+        allowNavigation = getArray(configJSON, "server.allowNavigation", null);
+
+        // Android
+        overriddenUserAgentString = getString(configJSON, "android.overrideUserAgent", getString(configJSON, "overrideUserAgent", null));
+        appendedUserAgentString = getString(configJSON, "android.appendUserAgent", getString(configJSON, "appendUserAgent", null));
+        backgroundColor = getString(configJSON, "android.backgroundColor", getString(configJSON, "backgroundColor", null));
+        allowMixedContent =
+            getBoolean(configJSON, "android.allowMixedContent", getBoolean(configJSON, "allowMixedContent", allowMixedContent));
+        captureInput = getBoolean(configJSON, "android.captureInput", captureInput);
+        webContentsDebuggingEnabled = getBoolean(configJSON, "android.webContentsDebuggingEnabled", webContentsDebuggingEnabled);
+        hideLogs = getBoolean(configJSON, "android.hideLogs", getBoolean(configJSON, "hideLogs", hideLogs));
+
+        // Plugins
+        pluginConfigurations = getObject(configJSON, "plugins");
+    }
+
+    public boolean html5mode() {
```

Comment:
I think this should follow the getter pattern: `getHTML5Mode()` or `isHTML5Mode()`